packages/odyssey-react/src/components/Table/TableSortHeaderCell.tsx

```diff
@@ -0,0 +1,76 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { useState, forwardRef } from "react";
+import type { ReactNode, ComponentPropsWithRef } from "react";
+import { withTheme } from "@okta/odyssey-react-theme";
+import type { CellTextFormats } from "./types";
+import { useCx, useOmit } from "../../utils";
+import styles from "./TableCell.module.scss";
+import { theme } from "./TableCell.theme";
+import { TableSortButton } from "./TableSortButton";
+
+export interface TableSortHeaderCellProps
+  extends Omit<ComponentPropsWithRef<"th">, "style" | "className"> {
+  children?: ReactNode;
+  /**
+   * The basic text format for the cell.
+   */
+  format?: CellTextFormats;
+  /**
+   * Screen read only text used as the call to action for the button
+   */
+  screenReaderCallToAction: string;
+
+  /**
+   * Callback executed when the sort header fires a change event
+   * @param {Object} event the event object
+   * @param {string} value the string value of the input
+   */
+  onSort?: (value: "asc" | "desc" | "unsorted") => void;
+}
+
+export const TableSortHeaderCell = withTheme(
+  theme,
+  styles
+)(
+  forwardRef<HTMLTableCellElement, TableSortHeaderCellProps>((props, ref) => {
+    const { children, format, screenReaderCallToAction, onSort, ...rest } = props;
+    const [direction, setDirection] = useState<"asc" | "desc" | "unsorted">("unsorted");
```

Comment:
you could directly use the `aria-sort` type here, I think.
```
const [direction, setDirection] = useState<AriaAttributes['aria-sort']>("none"); 
```
Could also update `TableSortButton`'s direction to be the same.